### PR TITLE
MODMS-29 JavaScript matchkeys/goldrush

### PR DESCRIPTION
**Pull request by dcrossleyau**
The `matchkeys/goldrush](matchkeys/goldrush/goldrush.js` implements the "Gold Rush - Colorado Alliance MARC record match key generation" (specification September 2021).

It takes input being a MARC-in-JSON string of MARC fields, and returns the matchkey string.

See [MODMS-29](https://issues.folio.org/browse/MODMS-29).

The README explains how to run test processing of the initial sample files. Add other various local samples with additional MARC fields.

### CODE REVIEW

File: js/matchkeys/goldrush/goldrush.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+// Returns matchkey as string.
+//
+// Version: 1.0.0 (for specification September 2021)
+
+function loadMarcJson(marcJson) {
+  let marcObj;
+  try {
+    marcObj = JSON.parse(marcJson);
+  } catch (e) {
+    throw new Error(e.message);
```
Comment (skoczko) [inline js/matchkeys/goldrush/goldrush.js:13]: Unless you're adding some additional information to the error, this seems redundant and you should just let the "parse" fail and the exception will propagate.

You can add some context like this: `throw new Error('Input must be JSON', { cause: e });`

File: js/test/main.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const testsPath = 'test/records';
+const files = fs.readdirSync(testsPath);
+const testFiles = files.filter((file) => path.extname(file) === '.json');
+for (let n = 0; n < testFiles.length; n += 1) {
+  const testFile = `${testsPath}/${testFiles[n]}`;
+  const marcJson = fs.readFileSync(testFile, 'utf8');
+  console.log(`Processing ${testFile}`);
+  const keyStr = goldrush.matchkey(marcJson);
+  console.log(`${keyStr}`);
```
Comment (skoczko) [inline js/test/main.js:13]: TODO: add asserts with expected values.

File: js/matchkeys/goldrush/goldrush.js (hunk 1 above)
Comment (dcrossleyau) [reply js/matchkeys/goldrush/goldrush.js]: Done.

File: js/test/main.js (hunk 2 above)
Comment (dcrossleyau) [reply js/test/main.js]: Done an initial set.

**Pull request merged**